packages/web3/src/web3.ts

```diff
@@ -102,7 +117,41 @@ export class Web3 extends Web3Context<EthExecutionAPI> {
 			}
 		}
 
-		const eth = self.use(Web3Eth);
+		const eth = new Proxy<Web3Eth>(self.use(Web3Eth), {
+			get(target: Web3Eth, prop: string) {
+				const originalMethod = target[prop as keyof Web3Eth];
+				if (
+					isQueryMethod(prop) &&
+					typeof originalMethod === 'function' &&
+					prop !== 'getBlockNumber'
+				) {
+					return async (...args: unknown[]): Promise<unknown> => {
+						let funcArgs: unknown[] = args;
+						if (!args.length) {
+							const latestBlockNumber: bigint = await target.getBlockNumber();
+							funcArgs = [latestBlockNumber];
+						}
+						const replicatedQueries = Array.from(
+							{ length: 4 },
+							async () =>
+								(originalMethod as QueryFunction<unknown[], unknown>).apply(
+									target,
+									funcArgs,
+								) as Promise<unknown>,
+						);
+						const promises = await Promise.all(replicatedQueries);
```

Comment:
This might fail if some of nodes is unable to query for some reason. might be better to use allSettled?

Reply:
I need all promises to succeed for the operation to be considered successful